_hardcoded/tracing.go

```diff
@@ -10,119 +10,149 @@ import (
 	"go.opentelemetry.io/otel"
-	"go.opentelemetry.io/otel/exporters/otlp"
-	"go.opentelemetry.io/otel/exporters/otlp/otlpgrpc"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/baggage"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
 	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	semconv "go.opentelemetry.io/otel/semconv/v1.10.0"
 	"go.opentelemetry.io/otel/trace"
-	"gopkg.in/Clever/kayvee-go.v6/logger"
 )
 
-// propagator to use.
-var propagator propagation.TextMapPropagator = propagation.TraceContext{} // traceparent header
-
-// defaultCollectorPort was changed from 55860 in November and the Go library
-// hasn't been updated when it is updated we can use otlp.DefaultCollectorPort
+var defaultCollectorHost string = "localhost"
 var defaultCollectorPort uint16 = 4317
 
-// SetupGlobalTraceProviderAndExporter sets up an exporter to export,
-// as well as the opentelemetry global trace provider for trace generators to use.
-// The exporter and provider are returned in order for the caller to defer shutdown.
 func SetupGlobalTraceProviderAndExporter(ctx context.Context) (sdktrace.SpanExporter, *sdktrace.TracerProvider, error) {
-	// 1. set up exporter
-	// 2. set up tracer provider
-	// 3. assign global tracer provider
 
-	// If we're running locally, then turn off sampling. Otherwise sample
-	// 1% or whatever TRACING_SAMPLING_PROBABILITY specifies.
-	samplingProbability := 0.01
-	isLocal := os.Getenv("_IS_LOCAL") == "true"
-	if isLocal {
-		samplingProbability = 1.0
-	} else if v := os.Getenv("TRACING_SAMPLING_PROBABILITY"); v != "" {
-		samplingProbabilityFromEnv, err := strconv.ParseFloat(v, 64)
-		if err != nil {
-			return nil, nil, fmt.Errorf("could not parse '%s' to float", v)
-		}
-		samplingProbability = samplingProbabilityFromEnv
-	}
-
-	addr := fmt.Sprintf("%s:%d", otlp.DefaultCollectorHost, defaultCollectorPort)
+	addr := fmt.Sprintf("%s:%d", defaultCollectorHost, defaultCollectorPort)
 
 	// Every 15 seconds we'll try to connect to opentelemetry collector at
 	// the default location of localhost:4317
 	// When running in production this is a sidecar, and when running
 	// locally this is a locally running opetelemetry-collector.
-	driver := otlpgrpc.NewDriver(
-		otlpgrpc.WithReconnectionPeriod(15*time.Second),
-		otlpgrpc.WithEndpoint(addr),
-		otlpgrpc.WithInsecure(),
-	)
-	exporter, err := otlp.NewExporter(
-		ctx,
-		driver,
+	otlpClient := otlptracegrpc.NewClient(
+		otlptracegrpc.WithReconnectionPeriod(15*time.Second),
+		otlptracegrpc.WithEndpoint(addr),
+		otlptracegrpc.WithInsecure(),
 	)
+	spanExporter, err := otlptrace.New(ctx, otlpClient)
+	// spanExporter, err := jaeger.New(jaeger.WithCollectorEndpoint(jaeger.WithEndpoint("http://localhost:14268/api/traces")))
```

Comment:
When would one uncomment this? Maybe add an extra note along the lines of "to test locally, you can use Jaeger and use this line instead of the previous"